src/util/Fs.cpp

```diff
@@ -442,6 +442,33 @@ checkNoGzipSuffix(std::string const& filename)
     }
 }
 
+std::streamsize
+size(std::ifstream& ifs)
+{
+    assert(ifs.is_open());
+
+    ifs.ignore(std::numeric_limits<std::streamsize>::max());
+    auto result = ifs.gcount();
+    ifs.clear();
+    ifs.seekg(0, std::ios_base::beg);
+
+    return result;
+}
+
+std::streamsize
+size(std::string const& filename)
+{
+    struct stat st;
+    if (stat(filename.c_str(), &st) == 0)
```

Comment:
why the duplicate implementation? I am not sure this is portable. you can just open a stream and call the other one.

Reply:
I didn't want to open a file just to get its size, but I can do it.